src/mgmt/srv.c

```diff
@@ -300,6 +303,100 @@ process_join_request(Drpc__Call *drpc_req, Mgmt__JoinResp *resp)
 		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
 }
 
+static void
+process_create_pool_request(Drpc__Call *drpc_req, Mgmt__CreatePoolResp *resp)
+{
+	Mgmt__CreatePoolReq	*pb_req = NULL;
+	d_rank_list_t		*targets = NULL;
+	d_rank_list_t		*svc = NULL;
+	uuid_t			pool_uuid;
+	int			buflen = 16;
+	int			len;
+	int			i;
+	int			rc = 0;
+	char			*id = NULL;
+	char			*buf = NULL;
+	char			*extra = NULL;
+
+	/* response status is populated with SUCCESS on init */
+	mgmt__create_pool_resp__init(resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_req = mgmt__create_pool_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_req == NULL) {
+		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+		D_ERROR("Failed to extract request\n");
+
+		return;
+	}
+
+	/* parse targets rank list */
+	if (strlen(pb_req->ranks) != 0) {
+		targets = daos_rank_list_parse(pb_req->ranks, ",");
+		if (targets == NULL) {
+			D_ERROR("failed to parse target ranks\n");
+			D_GOTO(out, rc = -1);
```

Comment:
Although it doesn't strictly matter in this function, we usually use -DER_NOMEM in this kind cases, instead of an invalid error code -1.

Reply:
yes I specified -1 because it's only used locally and won't be returned